plugin/src/lib.rs

```diff
@@ -496,6 +496,29 @@ impl<'a, 'cx> Folder for MutatorPlugin<'a, 'cx> {
                     attrs,
                 })
             }
+            Expr {
+                id,
+                node: ExprKind::WhileLet(pat, cond, block, opt_label),
+                span,
+                attrs,
+            } => {
+                let (n, current, sym, flag, mask) = self.add_mutations(
+                        cond.span,
+                        &["replacing while condition with false"],
+                    );
+                let cond = self.fold_expr(cond);
+                let block = fold::noop_fold_block(block, self);
+                let mut_cond = quote_expr!(self.cx(), {
+                    ::mutagen::report_coverage($n..$current, &$sym[$flag], $mask);
+                    ::mutagen::w($cond, $n)
```

Comment:
I think this won't work. Taking this code as example:

```rust
    let mut v = vec![1, 2, 3, 4, 5];
    let mut i = v.iter_mut();

    while let Some(i) = i.next() {
        println!("{}", i);
    }
```

Will be mutated to:

```rust

    let mut v = vec![1, 2, 3, 4, 5];
    let mut i = v.iter_mut();

    while let Some(i) = if { self.now($mutation_count) { false } else { i.next() }  {
        println!("{}", i);
    }
```
This will lead that "if" will have mismatching types: on one branch will return a boolean while the other will return `Option<{integer}>`.

@llogiq If the goal is mutate `while let Some(n) = $e` into `while let Some(n) = None`, it will be the same as no executing the loop, no? Maybe we can just mutate it to `if false { $body }` instead?  I've got the feeling I'm missing something